### ISSUE TITLE
Migrate over to gh-pages from Heroku?

### ISSUE THREAD

**Issue by fr3fou** [taylorjg/rubiks-cube#1]
Hey!

I stumbled across your project randomly while looking for a 2D visualiser for scrambles for my own project (a React-based timer) and took a look at your code. I noticed that you are using Express to host a static webpage and was curious why did you decide to use Heroku's free plan (I can only assume that it is the free plan because it took my browser ~10-20 seconds to load - I'd assume that's the Dyno worker starting up). I went ahead and cloned your repo to try to run it on my machine using npm, but it turned out you .gitignore-d the public folder and I have no clue what it's in it. 

So my question is, do you actually do some sort of backend-logic on Heroku or is it all just a fully-static website? If no, then I think you can easily adjust your code / webpack config (haven't taken a deep look at myself) and not rely on Heroku, rather you can host it for free on GitHub-Pages (assuming you don't do any sort of backend logic, you can then easily address the "long" loading times). If you don't like GitHub-Pages, you could also use [Netlify](https://www.netlify.com/) which does have some bonuses over gh-pages.

Cheers! 

Simo Aleksandrov :3

**Comment by taylorjg** [comment]
Hi,

Yes it's hosted on the Heroku free plan. I have a bunch of projects that are hosted there. Some of them do have backend logic. I tend to just use the same approach for hosting regardless of whether there is backend logic or not so I don't have to think about it. For [this project ](https://github.com/taylorjg/leaving), I used gh-pages. I normally use CircleCI to automate deployment to Heroku from master. I looked into doing the same to gh-pages (see [here](https://github.com/DevProgress/onboarding/wiki/Using-Circle-CI-with-Github-Pages-for-Continuous-Delivery)) but it seemed a real faff so I decided to stick with Heroku.

Re the `public` folder - this folder is built by webpack. Use the following command to do a build:

```
node_modules/.bin/webpack
```

or if you have npx installed:

```
npx weback
```

Regards,
Jon
